src/System.Private.CoreLib/shared/System/IO/MemoryStream.cs

```diff
@@ -167,7 +163,7 @@ private bool EnsureCapacity(int value)
                 // And we want to give the user the value that they asked for
                 if ((uint)(_capacity * 2) > Array.MaxByteArrayLength)
                 {
-                    newCapacity = value > Array.MaxByteArrayLength ? value : Array.MaxByteArrayLength;
+                    newCapacity = Math.Min(value, Array.MaxByteArrayLength);
```

Comment:
Shouldn't this be max, not min?

Reply:
Actually, thinking about it shouldn't this throw? We set the value to "unsupported" range and hit GC on the next call to deal with it. What type of exception are runtimes expected to throw for this case (the size bigger than MaxByteArrayLength) ?

Reply:
> Actually, thinking about it shouldn't this throw? 

It will, when it tries to do the array allocation that's larger than allowed. That's why the max here is important.

Reply:
What I meant is to throw from here e.g. explicit OutOfMemoryException with full stack trace instead of asking GC to allocate not supported size.

Reply:
From my perspective, all of this logic is just a heuristic to determine how much memory to ask for... the actual check for whether the request is fulfillable comes from the allocator, which is the authority on the matter.